### DISCUSSION TITLE
linting & formatting dilemma

### DISCUSSION THREAD

**Discussion by cStralpt** [oxc-project/oxc#2102] [General]
i try to understand about  what are these 2 tools claim,
- **biome has 35x faster formatting code compared to prettier** while **oxlint is 50-100x faster linting speed compared to eslint**
- biome doesn't give linting speed benchmark?
- oxlint doesn't give formatting speed benchmark?
- does oxlint also do formatting? what it means it support prettier?
![image](https://github.com/oxc-project/oxc/assets/95400822/f12f6068-d8c3-4aa5-9c29-cc264ac871b9)
![image](https://github.com/oxc-project/oxc/assets/95400822/b629e4da-c921-44cd-ba0f-42e3a6c07e0f)

i hope there are very clear explanation here, why we should stick with which one of those.

**Comment by SYED-HUZAIFA-HASSAN** [comment]
Hi @cStralpt, hope you found your answer. However if you didn't I'll give you a quick rundown, oxc and biome are entirely different project with very different goals in mind. Biome is a fork of Rome, which was always set out to be linter + formatter bundled in one, it's aim was to improve overall linting + formatting performance with ESlint Compatibility an after-thought. However, oxc-project has subset of tools within the ecosystem, with aim to be a drop-in replacement for existing tools. For example, ox-lint only targets ESLint compatibility. Currently ox formatter is under development which targets preetier. Both projects have similar goals when you read a summary but they have different plans to ease adoption.

**EDIT:**

According to Current Notes of OX-Lint [here](https://oxc-project.github.io/docs/guide/usage/linter.html)

> At the current stage, oxlint is not intended to fully replace ESLint; it serves as an enhancement when ESLint's slowness becomes a bottleneck in your workflow.
> 
> We recommend running oxlint before ESLint in your lint-staged or CI setup for a quicker feedback loop, considering it only takes a few seconds to run on large codebases.

**Reply by raulmarindev** [reply]
This is the new url: https://oxc.rs/docs/guide/usage/linter.html. The old one seems to be redirecting to the home.

**Comment by Boshen** [comment]
We are implementing a formatter https://github.com/oxc-project/oxc/discussions/13608
